### PR TITLE
add pagination component to sparkle

### PR DESCRIPTION
**Pull request by camacrain**
## Description

Add pagination component to sparkle.

Screenshot 2024-06-17 at 9 39 54 AM
Screenshot 2024-06-17 at 9 40 28 AM

### CODE REVIEW

File: sparkle/src/components/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        size === "xs" ? "s-text-xs" : "s-text-sm"
+      )}
+      onClick={() => onPageClick(pageNumber)}
+    >
+        {pageNumber}
+    </button>
+  );
+}
+
+function renderElipses(size: "sm" | "xs") {
```
Comment (flvndvd) [inline sparkle/src/components/Pagination.tsx:33]: `nit`
```suggestion
function renderEllipses(size: "sm" | "xs") {
```

File: sparkle/src/components/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        "s-text-sm s-font-semibold s-text-slate-400",
+        size === "xs" ? "s-text-xs" : "s-text-sm"
+      )}
+    >
+      ...
+    </span>
+  );
+}
+
+function getPageButtons(currentPage: number, totalPages: number, slots: number, onPageClick: (currentPage: number) => void, size: "sm" | "xs") {
```
Comment (flvndvd) [inline sparkle/src/components/Pagination.tsx:46]: Let's extract the size in a proper type so you can re-use here.

File: sparkle/src/components/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+export function Pagination({
+  itemsCount,
+  maxItemsPerPage,
+  size = "sm",
+  showDetails = true,
+  showPageButtons = true
+}: PaginationProps) {
+  const numPages = Math.ceil(itemsCount / maxItemsPerPage);
+  const controlsAreHidden = numPages <= 1 ? true : false;
```
Comment (flvndvd) [inline sparkle/src/components/Pagination.tsx:106]: `nit`
```suggestion
  const controlsAreHidden = Boolean(numPages <= 1);
```

File: sparkle/src/components/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  const pagesShownInControls = 7;
```
Comment (flvndvd) [inline sparkle/src/components/Pagination.tsx:107]: Don't we want to condition this based on the size?

**Review by flvndvd — COMMENTED** [review]
\o/ Left a few comments.

**Review by Duncid — APPROVED** [review]

File: sparkle/src/components/Pagination.tsx (hunk 4 above)
Comment (flvndvd) [reply sparkle/src/components/Pagination.tsx:109]: Let's just move this const variables at the top of the file then 🙃.

**Pull request merged**